GitCommands/Settings/Setting.cs

```diff
@@ -4,91 +4,131 @@
 
 namespace GitCommands.Settings
 {
-    public abstract class Setting<T>
+    public interface ISetting<T>
     {
-        public readonly SettingsPath SettingsSource;
-        public readonly T DefaultValue;
-        public readonly string Name;
+        event EventHandler Updated;
 
-        public event EventHandler Updated;
+        SettingsPath SettingsSource { get; }
 
-        protected Setting(string name, SettingsPath settingsSource, T defaultValue)
+        string Name { get; }
+
+        T DefaultValue { get; }
+
+        T ValueOrDefault { get; }
+
+        T Value { get; set; }
+
+        string FullPath { get; }
+
+        bool ValueIsEmpty(T value);
```

Comment:
How about this?
```suggestion
        bool IsUnset(T value);
```

Reply:
Changed to "private IsDefault"
Checking for value types, in case got null